### PR TITLE
implement ApplyLabelFilter

### PR DESCRIPTION
**Pull request by IlyaFaer**
IPR: 1700

The docs I've managed to find:
https://cloud.google.com/bigtable/docs/reading-data#reading-filters
https://cloud.google.com/bigtable/docs/using-filters#apply-label
https://cloud.google.com/bigtable/docs/reference/data/rpc/google.bigtable.v2#cellchunk

### CODE REVIEW

File: bigtable/filter.go [hunk 1]
```diff
@@ -144,6 +144,18 @@ func (lnf latestNFilter) proto() *btpb.RowFilter {
...
+func ApplyLabelFilter(label string) Filter { return applyLabelFilter(label) }
+
+type applyLabelFilter string
+
+func (alf applyLabelFilter) String() string { return fmt.Sprintf("apply_label(%s)", string(alf)) }
+
+func (alf applyLabelFilter) proto() *btpb.RowFilter {
+	return &btpb.RowFilter{Filter: &btpb.RowFilter_ApplyLabelTransformer{ApplyLabelTransformer: string(alf)}}
+}
+
```
Comment (IlyaFaer) [inline bigtable/filter.go:158]: Adding a function to build a filter from protobufs
The filter is using only single one label due to [limitation](https://github.com/googleapis/go-genproto/blob/053ba62fc06f7e7fd79767598c2e73fbae71e745/googleapis/bigtable/v2/data.pb.go#L1262-L1263)

File: bigtable/integration_test.go [hunk 2]
```diff
@@ -880,6 +882,14 @@ func TestIntegration_Read(t *testing.T) {
...
 			want:   "gwashington-jadams-1,jadams-tjefferson-1",
 		},
+		{
+			desc:       "apply labels to the result rows",
+			rr:         RowRange{},
+			filter:     ApplyLabelFilter("test-label"),
+			limit:      LimitRows(2),
+			want:       "gwashington-jadams-1,jadams-gwashington-1,jadams-tjefferson-1",
+			wantLabels: []string{"test-label"},
+		},
```
Comment (IlyaFaer) [inline bigtable/integration_test.go:892]: Adding a test case. All the resulting rows must get the given label

File: bigtable/reader.go [hunk 3]
```diff
@@ -138,13 +140,15 @@ func (cr *chunkReader) handleCellValue(cc *btpb.ReadRowsResponse_CellChunk) Row
...
 			cr.curVal = make([]byte, 0, cc.ValueSize)
+			cr.curLabels = cc.Labels
 		}
 		cr.curVal = append(cr.curVal, cc.Value...)
 		cr.state = cellInProgress
 	} else {
 		// This cell is either the complete value or the last chunk of a split
 		if cr.curVal == nil {
 			cr.curVal = cc.Value
+			cr.curLabels = cc.Labels
```
Comment (IlyaFaer) [inline bigtable/reader.go:151]: According to the docs labels are sent in the first cell chunk. Thus, we're taking them when the first chunk of the cell is read

File: bigtable/reader.go [hunk 4]
```diff
@@ -165,9 +169,11 @@ func (cr *chunkReader) finishCell() {
 		Column:    string(cr.curFam) + ":" + string(cr.curQual),
 		Timestamp: Timestamp(cr.curTS),
 		Value:     cr.curVal,
+		Labels:    cr.curLabels,
 	}
 	cr.curRow[cr.curFam] = append(cr.curRow[cr.curFam], ri)
 	cr.curVal = nil
+	cr.curLabels = nil
```
Comment (IlyaFaer) [inline bigtable/reader.go:176]: When the cell is finished, we're dropping the labels

**Review by AlisskaPie — APPROVED** [review]

**Pull request closed without merging**